source/adapters/level_zero/memory.cpp

```diff
@@ -2156,58 +2154,22 @@ ur_result_t _ur_buffer::getBufferZeHandle(char *&ZeHandle,
     LastDeviceWithValidAllocation = Device;
     return UR_RESULT_SUCCESS;
   }
-  // Reads user setting on how to deal with buffers in contexts where
-  // all devices have the same root-device. Returns "true" if the
-  // preference is to have allocate on each [sub-]device and migrate
-  // normally (copy) to other sub-devices as needed. Returns "false"
-  // if the preference is to have single root-device allocations
-  // serve the needs of all [sub-]devices, meaning potentially more
-  // cross-tile traffic.
-  //
-  static const bool SingleRootDeviceBufferMigration = [] {
-    const char *UrRet =
-        std::getenv("UR_L0_SINGLE_ROOT_DEVICE_BUFFER_MIGRATION");
-    const char *PiRet =
-        std::getenv("SYCL_PI_LEVEL_ZERO_SINGLE_ROOT_DEVICE_BUFFER_MIGRATION");
-    const char *EnvStr = UrRet ? UrRet : (PiRet ? PiRet : nullptr);
-    if (EnvStr)
-      return (std::stoi(EnvStr) != 0);
-    // The default is to migrate normally, which may not always be the
-    // best option (depends on buffer access patterns), but is an
-    // overall win on the set of the available benchmarks.
-    return true;
-  }();
 
   // Peform actual device allocation as needed.
   if (!Allocation.ZeHandle) {
-    if (!SingleRootDeviceBufferMigration && UrContext->SingleRootDevice &&
```

Comment:
did you intentionally remove this logic?
why?

Reply:
`SingleRootDevice` from Context is default initialized to `nullptr` and is never set to any value, hence this if statement would always evaluate to `false` so I've removed this variable alltogether.